ImageInLib/ImageInLib/src/fast_sweeping_method.c

```diff
@@ -16,8 +16,11 @@ bool fastSweepingFunction_3D(dataType ** distance3DPtr, dataType ** curve3DPtr,
 	size_t sweepNumber = 0, sweepDirection;
 	const size_t dim2D = xDim * yDim;
 	dataType ** temp3dPtr = (dataType **)malloc(sizeof(dataType*) * zDim);// temporary array used in the computation of distance
-	for (i = 0; i < zDim; i++)
+	dataType ** tmpcurve3DPtr = (dataType **)malloc(sizeof(dataType*) * zDim);// temporary array to copy curve3DPtr
+	for (i = 0; i < zDim; i++) {
 		temp3dPtr[i] = (dataType *)malloc(sizeof(dataType) * dim2D);
+		tmpcurve3DPtr[i] = (dataType *)malloc(sizeof(dataType) * dim2D);
```

Comment:
Just optimization issue - sizeof(dataType) * dim2D could be extracted as const variable and used where it is necessary (do not evaluate it each time)